packages/gatsby/src/bootstrap/load-plugins/load.ts

```diff
@@ -233,15 +233,20 @@ export function loadPlugins(
 
   // TypeScript support by default! use the user-provided one if it exists
   const typescriptPlugin = (config.plugins || []).find(
-    plugin =>
-      (plugin as IPluginRefObject).resolve === `gatsby-plugin-typescript` ||
-      plugin === `gatsby-plugin-typescript`
```

Comment:
Note: removed the string case because `plugin` _cannot_ be a string at this point anymore now that we normalize them ahead of time. TypeScript helpfully pointed that out to me!